src/python/dxpy/scripts/dx.py

```diff
@@ -2803,6 +2804,21 @@ def run_body(args, executable, dest_proj, dest_path, preset_inputs=None, input_n
         args.instance_type = dict({stage: reqs['instanceType'] for stage, reqs in list(args.sys_reqs_from_clone.items())},
                                   **(args.instance_type or {}))
 
+    if args.sys_reqs_from_clone and not isinstance(args.instance_count, str):
+        # extract instance counts from cloned sys reqs and override them with args provided with "dx run"
+        args.instance_count = dict({fn: reqs['clusterSpec']['initialInstanceCount']
+                                        for fn, reqs in list(args.sys_reqs_from_clone.items()) if 'clusterSpec' in reqs},
+                                   **(args.instance_count or {}))
```

Comment:
Would this work if `args.instance_count` was just a string versus a dict? Would it be possible to use SRD here?

Reply:
If the args.instance_count is a string, the wildcard entry point will used and the count needs to overwrite values entry points (so we don't care about cloned vals).

Both args.instance_count and args.instance_type set here from cloning could be set as SRD but it'd require some adjustments down below. The SRD for instance types is created later, in `dxapplet.__get_run_input_common_fields()` so that `instance_type` can be parsed in one place regardless whether we're getting it from CLI args or from the `executable.run()` method. The `run(instance_count)` is not implemented yet since it requires again some more heavy lifting (since we require app description to process `instance-count` properly and the `run()` needs some more reorg for that purpose, to maybe not call another `describe` if it was done before).

Reply:
```
>>> dict({"foo": 2}, **(None or {"bar": 4}))
{'foo': 2, 'bar': 4}
>>> dict({"foo": 2}, **("baz" or {"bar": 4}))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: type object argument after ** must be a mapping, not str
>>> dict({"foo": 2}, **({"foo": 4}  or {"bar": 4}))
{'foo': 4}
```

Reply:
Right, but in the condition we exclude a string:
`... and not isinstance(args.instance_count, str)`

We want to exclude it since a string is where a wildcard '*' needs to be applied, so we don't care about whatever values are in the system requirements from the cloned job - we'll overwrite it anyway.